test/addons/e10s-remote/main.js

```diff
@@ -0,0 +1,581 @@
+
+  let events = [];
+
+  // In e10s we should expect to see two processes
+  let expectedCount = isE10S ? 2 : 1;
+
+  yield new Promise(resolve => {
+    let count = 0;
+
+    // Wait for a process to be detected
+    let listener = process => {
+      console.log("Saw a process attach");
+      // Wait for the remote module to load in this process
+      process.port.once('sdk/test/load', () => {
+        console.log("Saw a remote module load");
+        count++;
+        if (count == expectedCount) {
+          processes.off('attach', listener);
+          resolve();
+        }
+      });
+    }
+    processes.on('attach', listener);
+  });
+
+  console.log("Remote ready");
+  return { processes, frames, remoteRequire };
+});
+
+if (isE10S) {
+  console.log("Testing in E10S mode");
+  // We expect a child process to already be present, make sure that is the case
+  mainWindow.XULBrowserWindow.forceInitialBrowserRemote();
+
+  // Check that we see a process stop and start
+  exports["test process restart"] = function*(assert) {
+    let window = getMostRecentBrowserWindow();
+
+    let tabs = getTabs(window);
+    assert.equal(tabs.length, 1, "Should have just the one tab to start with");
+    let tab = tabs[0];
+
+    let loader = new Loader(module);
+    let { processes, frames, remoteRequire } = yield waitForProcesses(loader);
+
+    let remoteProcess = Array.filter(processes, p => p.isRemote)[0];
+    let localProcess = Array.filter(processes, p => !p.isRemote)[0];
+    let remoteFrame = Array.filter(frames, f => f.process == remoteProcess)[0];
+
+    // Switch the remote tab to a local URI which should kill the remote process
+
+    let frameDetach = promiseEventOnItemAndContainer(assert, remoteFrame, frames, 'detach');
+    let frameAttach = promiseEvent(frames, 'attach');
+    let processDetach = promiseEventOnItemAndContainer(assert, remoteProcess, processes, 'detach');
+    setTabURL(tab, LOCAL_URI);
+    // The load should kill the remote frame
+    yield frameDetach;
+    // And create a new frame in the local process
+    let [newFrame] = yield frameAttach;
+    assert.equal(newFrame.process, localProcess, "New frame should be in the local process");
+    // And kill the process
+    yield processDetach;
+
+    frameDetach = promiseEventOnItemAndContainer(assert, newFrame, frames, 'detach');
+    processAttach = promiseEvent(processes, 'attach');
+    frameAttach = promiseEvent(frames, 'attach');
+    setTabURL(tab, REMOTE_URI);
+    // The load should kill the remote frame
+    yield frameDetach;
+    // And create a new remote process
+    [remoteProcess] = yield processAttach;
+    assert.ok(remoteProcess.isRemote, "Process should be remote");
+    // And create a new frame in the remote process
+    [newFrame] = yield frameAttach;
+    assert.equal(newFrame.process, remoteProcess, "New frame should be in the remote process");
+
+    setTabURL(tab, "about:blank");
+
+    loader.unload();
+    yield cleanUI();
```

Comment:
let's just add one of these at the end of the file to reduce the repetition:

    after(exports, function*(name, assert) {
      yield cleanUI();
    });